File: task2/src/python/main.py
import json
import os
from http.server import HTTPServer, BaseHTTPRequestHandler
from urllib.parse import urlparse, parse_qs


class AppHandler(BaseHTTPRequestHandler):
    def _send_json(self, data):
        body = json.dumps(data).encode()
        self.send_response(200)
        self.send_header("Content-Type", "application/json")
        self.end_headers()
        self.wfile.write(body)

    def do_GET(self):
        parsed = urlparse(self.path)
        if parsed.path == "/health":
            self._handle_health()
        elif parsed.path == "/hello":
            self._handle_hello(parse_qs(parsed.query))
        else:
            self.send_response(404)
            self.end_headers()

    def _handle_health(self):
        port = os.environ.get("PORT", "8080")
        self._send_json({
            "message": "Server is running",
            "status": "ok",
            "port": port,
        })

    def _handle_hello(self, query):
        name = query.get("name", ["World"])[0]
        if not name:
            name = "World"
        self._send_json({
            "message": f"Hello, {name}!",
            "status": "ok",
        })

    def log_message(self, format, *args):
        pass


def main():
    port = int(os.environ.get("PORT", "8080"))
    server = HTTPServer(("0.0.0.0", port), AppHandler)
    server.serve_forever()


if __name__ == "__main__":
    main()

File: task2/src/go/main.go
package main

import (
	"encoding/json"
	"fmt"
	"net/http"
	"os"
)

type Response struct {
	Message string `json:"message"`
	Status  string `json:"status"`
	Port    string `json:"port"`
}

func healthHandler(w http.ResponseWriter, r *http.Request) {
	port := os.Getenv("PORT")
	if port == "" {
		port = "8080"
	}
	resp := Response{
		Message: "Server is running",
		Status:  "ok",
		Port:    port,
	}
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(resp)
}

func helloHandler(w http.ResponseWriter, r *http.Request) {
	name := r.URL.Query().Get("name")
	if name == "" {
		name = "World"
	}
	resp := Response{
		Message: fmt.Sprintf("Hello, %s!", name),
		Status:  "ok",
	}
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(resp)
}

func main() {
	port := os.Getenv("PORT")
	if port == "" {
		port = "8080"
	}

	http.HandleFunc("/health", healthHandler)
	http.HandleFunc("/hello", helloHandler)

	http.ListenAndServe(":"+port, nil)
}
